Feature/includeable vcl

zghosts opened the pull request:
Split of the vcl's from the tests into the /config directory ready to be included in the varnish configuration.
These vcl files are still included in the unit tests and as such are part of the tests

ddeboer commented:
User context still needs to be done, right?

dbu commented on doc/varnish-configuration.rst:
```diff
@@ -88,23 +107,22 @@ Varnish contains a `ban lurker`_ that crawls the content to eventually throw out
-    The custom ``X-Cache-Tags`` header should match the tagging header
-    :ref:`configured in the cache invalidator <custom_tags_header>`.
+    If you want to use different tag from ``X-Cache-Tags`` as defined in fos_ban.vcl,
+    the header should match the tagging header :ref:`configured in the cache invalidator <custom_tags_header>`.
```
i think this is a bit confusing. i would say something like:

If you need to use a different tag for the headers than the default ``X-Cache-Tags`` used in ``fos_ban.vcl``,
you need to write your own VCL code and change the tagging header :ref:`configured in the cache invalidator <custom_tags_header>`.

dbu commented on doc/varnish-configuration.rst:
```diff
@@ -40,14 +40,19 @@ To configure Varnish for `handling PURGE requests <https://www.varnish-cache.org
-    .. literalinclude:: ../tests/Functional/Fixtures/varnish-3/purge.vcl
+    .. literalinclude:: ../tests/Functional/Fixtures/varnish-3/fos.vcl
         :language: varnish3
+        :lines: 17,19-20,25-32
```
dont you miss line 15 here?

i wonder if its worth it to literalinclude here or whether we should just copy the vcl here, as its very small.

ddeboer replied:
Yeah, was wondering the same. While I’m all in favour of including (tested) VCL in our docs, just those couple of lines may not be worth the effort.

dbu commented:
thanks a lot! i think we maybe want to add an introductionary section explaining the concept of the includable vcl and the options to use absolute paths or symlink or copy the files. and that you can copy the file and edit if you need customization, at the cost of no longer automatically updating.

dbu commented:
we can still do the user context later and already merge this i think, when the details have been sorted out here.

zghosts commented:
Hi David's
I extracted the user_context.vcl.
Fixed the documentation errors
Updated the documentation.
I moved the vcl config directory into a resouces directory , don't know if you agree with that, but it makes more sense to me to put it there.

I also fixed a small docblock error

ddeboer commented:
:+1:

I agree with the resources directory. Can you please squash commits?

zghosts commented:
Resubmitted in #229

ddeboer commented:
Just a tip: for a PR, you can squash (rebase) commits and force push them to the PR branch, so you won't have to open a separate PR. :wink:
